src/ws-tab-menu/ws-tab-menu.js

```diff
@@ -0,0 +1,165 @@
+import {React, Component, PropTypes} from '../imports';
+
+/**
+ * This component renders a tab menu and provides the selected tab via a value property.
+ * @property {Array} items List of menu items {value: any & required, label: string, className: string, badge: string}
+ * @property {*} value Matching a value of an item
+ * @property {Function} onChange Callback function for React/Preact applications
+ */
+export class WSTabMenu extends Component {
+
+  static propTypes = {
+    items: PropTypes.array,
+    value: PropTypes.any,
+    onChange: PropTypes.func
+  };
+
+  static defaultProps = {
+    items: [],
+    value: null,
+    onChange: null
+  };
+
+  /**
+   * @param {Object} props React component properties
+   */
+  constructor(props) {
+    super(props);
+
+    this.state = this.createState(props);
+    this.menuItems = [];
+  }
+
+  /**
+   * Bind listeners and move dash below active item
+   * @returns {void}
+   */
+  componentDidMount() {
+    this.menuItems.forEach((element, index) => {
+      const item = this.props.items[index];
+      // Move dash below selected item
+      if (item.value === this.state.value) {
+        this.animateDash(element);
+      }
+      element.addEventListener('click', this.onClick);
+    });
+  }
+
+  /**
+   * Update state from property change
+   * @param {Object} props React properties
+   * @returns {void}
+   */
+  componentWillReceiveProps(props) {
+    this.setState(this.createState(props));
```

Comment:
From my humbled understanding should not rely directly from props as both are updated async (https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous) the save thing (we are also using in our new project) is to use an arrow function instead especially as on the next line we do a check with this.state.value and this might be another value then the one you save with setState ?

Reply:
This problem only appears if I rely on an old state to calculate the new one. But in this component I just take the changed props and generate a completely new state. Therefor I would say it is ok. If I'm wrong or I misunderstood something I would be happy to talk about a better solution :)